### PR TITLE
feat: Return bootstrap peers as iterator and filter relevant ones

### PR DESCRIPTION
**Pull request by zvolin**

### CODE REVIEW

**Comment by oblique** [comment]
Is this for minimizing the "not supported" logs? Or it fixes something else too?

**Comment by zvolin** [comment]
it's mostly the UX change I'd say, you are presented with many bootstrap peers for each network but they are all not working. That way we can show the true state to the user

File: celestia/src/server.rs [hunk 1]
```diff
@@ -53,6 +53,8 @@ pub(crate) async fn run(args: Params) -> Result<()> {
     let genesis_hash = network_genesis(network);
     let bootnodes = if args.bootnodes.is_empty() {
         canonical_network_bootnodes(network)
+            .filter(|addr| addr.protocol_stack().any(|proto| proto == "webtransport"))
```
Comment (oblique) [inline celestia/src/server.rs:56]: There is a type-safe way for this check:

```suggestion
            .filter(|addr| addr.iter().any(|proto| proto == Protocol::WebTransport))
```

File: node-wasm/src/node.rs [hunk 2]
```diff
@@ -197,7 +197,7 @@ impl WasmNodeConfig {
             network,
             genesis_hash: network_genesis(network.into()).map(|h| h.to_string()),
             bootnodes: canonical_network_bootnodes(network.into())
-                .iter()
+                .filter(|addr| addr.protocol_stack().any(|proto| proto == "webtransport"))
```
Comment (oblique) [inline node-wasm/src/node.rs:200]: same here

File: celestia/src/server.rs (hunk 1 above)
Comment (zvolin) [reply celestia/src/server.rs]: done

**Review by oblique — APPROVED** [review]

**Review by fl0rek — APPROVED** [review]
🎸

**Pull request merged**